flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskManagerServices.java

```diff
@@ -198,9 +197,9 @@ public static TaskManagerServices fromConfiguration(
 
 		final FileCache fileCache = new FileCache(taskManagerServicesConfiguration.getTmpDirPaths());
 
-		final List<ResourceProfile> resourceProfiles = new ArrayList<>(taskManagerServicesConfiguration.getNumberOfSlots());
+		final List<ResourceProfile> resourceProfiles = taskManagerServicesConfiguration.getResourceProfiles();
 
-		for (int i = 0; i < taskManagerServicesConfiguration.getNumberOfSlots(); i++) {
+		for (int i = resourceProfiles.size(); i < taskManagerServicesConfiguration.getNumberOfSlots(); i++) {
 			resourceProfiles.add(new ResourceProfile(1.0, 42L));
```

Comment:
Now only yarn resource manager will add resource profiles in the config, for standalone, there are none. so the resourceProfiles.size() will be zero, still need to add a ResourceProfile here. I think adding one can match all other resource profile is a good idea, I will introduce it.